lxd/instance/drivers/driver_qemu.go

```diff
@@ -7061,6 +7062,50 @@ func (d *qemu) checkFeatures(hostArch int, qemuPath string) ([]string, error) {
 		features = append(features, "cpu_hotplug")
 	}
 
+	// Check AMD SEV features (only for x86 architecture)
+	if hostArch == osarch.ARCH_64BIT_INTEL_X86 {
+		cmdline, err := os.ReadFile("/proc/cmdline")
+		if err != nil {
+			return nil, err
+		}
+
+		parts := strings.Split(string(cmdline), " ")
+
+		// Check if SME is enabled in the kernel command line.
+		if slices.IndexFunc(parts, func(p string) bool { return p == "mem_encrypt=on" }) != -1 {
+			features = append(features, "sme")
+		}
+
+		// Check if SEV/SEV-ES are enabled
+		sev, err := os.ReadFile("/sys/module/kvm_amd/parameters/sev")
+		if err != nil {
+			logger.Debug("Failed reading SEV capability during VM feature check", logger.Ctx{"err": err})
+		} else {
+			if string(sev) == "Y\n" {
+				// Host supports SEV, check if QEMU supports it as well.
+				capabilities, err := monitor.SEVCapabilities()
+				if err != nil {
+					logger.Debug("Failed querying SEV capability during VM feature check", logger.Ctx{"err": err})
+				} else {
+					features = append(features, "sev")
+					features = append(features, fmt.Sprintf("sev.cbitpos=%d", capabilities.CBitPos))
```

Comment:
Hrm, I'm not really enjoying the `feature=value` encoding technique we are introducing here, as it then requires decoding elsewhere. This is something that (with hindsight) we also mistakenly fell into with the RunConfig shared between device driver and instance driver, and ended up having to do use custom (undocumented) encoding of various things.

Because these values are just used internally, I suggest we change the features slice into a `map[string]any` (this would need to be a series of commits earlier in this PR or a separate PR before introducing the SEV concept). 

Then for features that have no additional info the properties, it can just store the name of the feature and nil as the value, and then for features that do have additional properties they can be stored as the actual data type that `monitor.SEVCapabilites` returns.

Checking for a feature and then accessing the feature options then becomes easy as one can do something like this:

```
featureOpts, found := features["foo"]
if found {
   fooFeatureOpts, ok := featureOpts.(dataType)
  if !ok {
      // Unexpected invalid data type.
  }
}
```

Reply:
That way we don't have to do any manual encoding/decoding, and we benefit from Go's type checking, as well as being able to more easily reference where these data types are used.